core/display.js

```diff
@@ -224,6 +224,14 @@ export default class Display {
         this.viewportChangePos(0, 0);
     }
 
+    copyImageData() {
+        return this._targetCtx.getImageData(0, 0, this.width, this.height)
+    }
+
+    toDataURL(format) {
```

Comment:
Let's mimic the API fully and also include `encoderOptions`.

On the same theme, let's keep the same argument names as the canvas API and rename this to `type`.